src/unix/linux_like/android/mod.rs

```diff
@@ -602,22 +604,31 @@ cfg_if! {
             }
         }
 
+        #[deprecated(
+            note = "this trait implementation is unsound and will be removed"
+        )]
```

Comment:
The attr should be applied to struct, not impl. Could you also add `since = "0.2.80"`?

Reply:
Done. I also had to add some `#[allow(deprecated)]` in `src/macros.rs` to avoid warnings.